hbase_sql/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java

```diff
@@ -453,19 +458,20 @@ protected boolean performCompaction(Compactor.FileDetails fd, InternalScanner sc
                 if (closeCheckInterval > 0) {
                     bytesWritten += len;
                     if (bytesWritten > closeCheckInterval) {
-                        bytesWritten = 0;
-//                        if (!store.areWritesEnabled()) {
-//                            progress.cancel();
-//                            return false;
-//                        }
+                        bytesWritten = 0; // reset so check whether cancellation is requested in the next
+                                          // <closeCheckInterval>-bytes mark
+                        if (!store.areWritesEnabled()) { // this call could be expensive.
```

Comment:
This might always return false when running in Spark, can you verify that?

Reply:
Sure, I am on it.

Reply:
You're right, when `performCompaction` is called from Spark executor, the status of regions is always false causing compactions to be cancelled all the time.
To solve this, I am adding a `isSpark` flag that is set when running compactions from Spark, we use this flag to make sure we detect region status only when `performCompaction` is called from HBase and not Spark.
Thanks for the hint.

Reply:
... I added a branch for Spark jobs, if task is interrupted we perform similar cancellation preventing Spark executors from pointlessly running the compaction job.